lte/gateway/c/session_manager/LocalEnforcer.cpp

```diff
@@ -1046,18 +1051,8 @@ void LocalEnforcer::init_session(
 
 void LocalEnforcer::add_ue_to_shard(
     const std::string imsi, SessionMap& session_map, SessionState& session) {
-  auto sm_it     = session_map.find(imsi);
-  auto& sessions = sm_it->second;
-  // Check if UE has other sessions and reuse shard_id if so
-  for (auto session_it = sessions.begin(); session_it != sessions.end();
-       ++session_it) {
```

Comment:
This approach caused bugs in the polling, so I reverted to just checking if the UE already exists within the add method itself.

Reply:
What kind of bugs?

Reply:
It wasn't properly finding the session ids or setting them. Just seemed simpler to handle the case inside, without much performance compromise.